docs/reference/contrib/modeladmin/reversing_urls.rst

```diff
@@ -0,0 +1,126 @@
+determine what these URLs are.
+
+You can take a similar approach in your own code too, by creating a
+``ModelAdmin`` instance yourself, and using its ``url_helper``
+to determine URLs.
+
+See below for some examples:
+
+.. contents::
+    :local:
+    :depth: 1
+
+-------------------------------------------------------------------
+Getting the ``edit`` or ``delete`` or ``inspect`` URL for an object
+-------------------------------------------------------------------
+
+In this example, we will provide a quick way to ``edit`` the Author that is
+linked to a blog post from the Admin Page Listing menu. We have defined
+an ``AuthorModelAdmin`` class and registered it with Wagtail to allow
+``Author`` objects to administered via the admin area. The ``BlogPage`` model
+has an ``author`` field (a ``ForeignKey`` to the ``Author`` model) to allow a
+single author to be specified for each post.
+
+.. code-block:: python
+
+    # file: wagtail_hooks.py
+
+    from wagtail.admin.widgets import PageListingButton
+    from wagtail.contrib.modeladmin.options import ModelAdmin, modeladmin_register
+    from wagtail.core import hooks
+
+    # Author & BlogPage model not shown in this example
+    from models import Author
+
+    # ensure our modeladmin is created
+    class AuthorModelAdmin(ModelAdmin):
+        model = Author
+        menu_order = 200
+
+    # Creating an instance of `AuthorModelAdmin`
+    author_modeladmin = AuthorModelAdmin()
+
+    @hooks.register('register_page_listing_buttons')
+    def add_author_edit_buttons(page, page_perms, is_parent=False):
+        """
+        For pages that have an author, add an additional button to the page listing,
+        linking to the 'edit' page for that author.
+        """
+        author_id = getattr(page, 'author_id', None)
+        if author_id:
+            # the url helper will return something like: /admin/my-app/author/edit/2/
+            author_edit_url = author_modeladmin.url_helper.get_action_url('edit', author_id)
+            yield PageListingButton('Edit Author',  author_edit_url, priority=10)
+
+    modeladmin_register(AuthorModelAdmin)
+
+
+In the above example, the ``url_helper`` on the ``AuthorModelAdmin`` class has
+been used to generate a URL to edit a ``Author`` instance via ``modeladmin``.
+
+The ``get_action_url`` function should be called with an action name such as
+``'edit'`` or ``'delete'`` and with the primary key (ID) of the instance.
+This will return a URL to load that action for the specific model in the format
+``/admin/my-app/my-page-model/edit/123``.
+
+The action names for a specific instance URLs are:
+
+* ``'edit'``
+* ``'delete'``
+* ``'inspect'`` Note: Requires inspect enabled in modeladmin configuration.
+
+
+.. note::
+    If your Primary Key for the model is likely to contain characters that
+    are not URL safe, ensure you wrap any data in ``quote`` from
+    ``django.contrib.admin.utils``.
```

Comment:
I don't think the title casing of "Primary Key" is really needed here (lower case is fine) :) Also, I think we can be a touch more precise about what data must be wrapped. Would suggest something like:

```
If you are using string values as primary keys for you model, and it's possible key values could contain characters that are not URL safe, you should use ``django.contrib.admin.utils.quote()`` to prepare primary key values before passing them to ``get_action_url()``. Failure to do this may result in Wagtail failing to recognise the primary key when the URL is visited, resulting in a 404 error response.
```